app/services/twilio_api_service.rb

```diff
@@ -86,19 +87,6 @@ def send_twilio_message(sender_number, recipient_number, message, callback_url =
       body: message
     )
   rescue Twilio::REST::TwilioError => exception
-    @error = exception
-    report_error(exception)
-  end
-
-  def report_error(e)
-    Sentry.capture_message(
-      "Error while processing notification",
-      extra: {
-        exception: e.to_s
-      },
-      tags: {
-        type: "notification-service"
-      }
-    )
+    raise Error.new("Error while calling the Twilio API", exception_message: exception)
```

Comment:
Ruby will wrap the original exception automatically for it and maintain that as the `cause` on our Error:

```suggestion
    raise Error, "Error while calling Twilio API"
  ```